cvat-data/src/ts/cvat-data.ts

```diff
@@ -239,8 +272,15 @@ export class FrameDecoder {
                     const height = Math.round(this.renderHeight / scaleFactor);
                     const width = Math.round(this.renderWidth / scaleFactor);
 
-                    const array = new Uint8ClampedArray(e.data.buf.slice(0, width * height * 4));
-                    createImageBitmap(new ImageData(array, width)).then((bitmap) => {
+                    createImageBitmap(FrameDecoder.cropImage(
+                        e.data.buf,
+                        e.data.width,
+                        e.data.height,
+                        0,
+                        0,
```

Comment:
- We do not need constant parameters.
- Function needs to be rewritten based on this constant values (code may be reduced). Is not it?

Reply:
Thanks for the response, I'll look into it.

Reply:
@bsekachev please have a look now.